From 0e3b76f8feb77b6cc340d20cbade69b234bb59e7 Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Mon, 11 Nov 2019 14:18:43 -0800
Subject: [PATCH 411/767] drivers: spi: octeontx2: set tritx in config register

commit 6b3ded8998edbef35b39b7dab0dfab884b2c08fe from
git@git.assembla.com:cavium/WindRiver.linux.git

With 9xx chips, bus width is expanded from x1 to x4, in order
to avoid driving of IO1/2/3 during bus idle, tritx must be
set to 1 to tristate these signals otherwise with some flash
parts, like Micron/Cypress, IO3 has multiplexed functionality
of HOLD and treats driving low as valid causing incorrect bus
operations to the flash parts.

Change-Id: Ia6c1f9eaa7cd86a63eadaf5f715f1f459863b384
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18707
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/spi/spi-octeontx2.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-octeontx2.c b/drivers/spi/spi-octeontx2.c
index 7ffb9f083c92..1470fed85443 100644
--- a/drivers/spi/spi-octeontx2.c
+++ b/drivers/spi/spi-octeontx2.c
@@ -81,6 +81,7 @@ static int octeontx2_spi_do_transfer(struct octeontx2_spi *p,
 	mpi_cfg.s.wireor = (mode & SPI_3WIRE) ? 1 : 0;
 	mpi_cfg.s.idlelo = cpha != cpol;
 	mpi_cfg.s.cslate = cpha ? 1 : 0;
+	mpi_cfg.s.tritx = 1;
 	mpi_cfg.s.enable = 1;
 	mpi_cfg.s.cs_sticky = 1;
 	mpi_cfg.s.legacy_dis = 1;
-- 
2.31.1

